### DISCUSSION TITLE
Component lifecycle

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#1096] [Proposals]
Existen casos en los que determinados componentes se disponibilizan antes de que exista una implementación de los mismos dentro de las librerías de diseño. 

El escenario ideal es que la definición de estos componentes no se modifique, pero existen casos en los que esto es necesario debido a:

* Cambios en tiempo de implementación
* Faseado de trabajo debido a limitaciones de tiempo
* Errores detectados con el uso del componente

Puede resultar valioso disponibilizar de alguna manera esta información para que cuando se usen, se tengan en cuenta los siguientes aspectos:

* Está sujeto a breaking changes en sus propiedades
* La definición del componente puede tener variaciones con respecto a su planteamiento inicial

Una manera de hacerlo sería indicar que esos recursos están en una fase beta. Para considerarlo como tal deberían cumplirse los siguientes requisitos:

**`Beta`**  
* El componente sólo vive en la librería de diseño / kit 
* No ha sido implementada en desarrollo su definición total (en el caso de que se haya faseado su desarrollo)
* No ha pasado una major release desde que se publicó

Aquí consideraría también otra fase (**`Alpha`** ?) en relación a aquellos elementos que:

* Están disponibilizados como módulos o ejemplos pero que se planea que tengan cabida dentro de las librerías core.
* Aún no existe un componente en las librerías pero su definición está terminada en las especificaciones de diseño.

**Comment by yceballost** [comment]
IMO, una etiqueta `alpha` o `beta` debería limitarse únicamente a elementos considerados componentes. Por lo que no incluiría en este lifecycle de componentes aquellas construcciones atómicas.

Pensando en alto, pondriamos considerar por ejemplo la poster card en el estado actual como `alpha`

Alpha: Componentes disponibles en diseño (specs hechas) pero no disponible en desarrollo
Beta: Componentes disponibles en diseño y desarrollo pero incompletos en desarrollo.

Ojo, el caso del hero es diferente, el hero actualmente no tendría estas etiquetas porque tanto en diseño como desarrollo está igual, si añadieramos entonces la siguiente fase a diseño.. aquí aplicaríamos estas etiquetas?

**Reply by aweell** [reply]
Estoy de acuerdo con sacar aquello que no es componente del saco. Mi comentario iba mas por aquello que es atómico pero ya se ha decidido que se va a incluir en core, pero el tiempo, puede dilatarse.

El caso del hero en concreto, que ocurre si algún aspecto de la fase 2 obliga a revisitar lo incluido en la fase uno y requiere de su modificación? No sería esto a valorar como: Hasta que no esté implementado todo lo que se planteo desde un inicio que ese componente va a soportar, no se considera distinto de `beta`.

**Reply by yceballost** [reply]
> Estoy de acuerdo con sacar aquello que no es componente del saco. Mi comentario iba mas por aquello que es atómico pero ya se ha decidido que se va a incluir en core, pero el tiempo, puede dilatarse.

mmm entiendo, yo propongo agarrar más ese estado a unas specs que a "esa decisión" que como bien dices, se ~puede dilatar~ dilatará en el tiempo y puede variar bastante.. sin embargo si se hicieron una specs es porque esa decisión ya maceró y se asentó.

> El caso del hero en concreto, que ocurre si algún aspecto de la fase 2 obliga a revisitar lo incluido en la fase uno y requiere de su modificación? No sería esto a valorar como: Hasta que no esté implementado todo lo que se planteo desde un inicio que ese componente va a soportar, no se considera distinto de beta.

quizás, y si consideramos que el componente es suceptible de ser evolucionado muy pronto debería llevar esta etiqueta? no me mola la idea de que sea una definición "al vuelo"...
